nifi-nar-bundles/nifi-rethinkdb-bundle/nifi-rethinkdb-processors/src/main/java/org/apache/nifi/processors/rethinkdb/PutRethinkDB.java

```diff
@@ -0,0 +1,237 @@
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        if ( flowFile.getSize() == 0) {
+            getLogger().error("Empty message");
+            flowFile = session.putAttribute(flowFile, RETHINKDB_ERROR_MESSAGE, "Empty message size " + flowFile.getSize());
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        if ( flowFile.getSize() > maxDocumentsSize) {
+            getLogger().error("Message size exceeded {} max allowed is {}", new Object[] { flowFile.getSize(), maxDocumentsSize});
+            flowFile = session.putAttribute(flowFile, RETHINKDB_ERROR_MESSAGE, "Max message size exceeded " + flowFile.getSize());
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        Charset charset = Charset.forName(context.getProperty(CHARSET).evaluateAttributeExpressions(flowFile).getValue());
+        String conflictStrategy = context.getProperty(CONFLICT_STRATEGY).evaluateAttributeExpressions(flowFile).getValue();
+        String durability = context.getProperty(DURABILITY).evaluateAttributeExpressions(flowFile).getValue();
+
+        try {
+            long startTimeMillis = System.currentTimeMillis();
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            session.exportTo(flowFile, baos);
+            String documents = new String(baos.toByteArray(), charset);
+            JSONParser parser = new JSONParser();
+            Object jsonDocuments = parser.parse(documents);
+
+            Insert insert = getRdbTable().insert(jsonDocuments)
+                .optArg(CONFLICT_OPTION_KEY, conflictStrategy)
+                .optArg(DURABILITY_OPTION_KEY, durability);
+
+            HashMap<String,Object> result = runInsert(insert);
+            final long endTimeMillis = System.currentTimeMillis();
+            getLogger().debug("Json documents {} inserted Result: {}", new Object[] {documents, result});
+            flowFile = populateAttributes(session, flowFile, result);
+
+            if ( (Long)result.get(RESULT_ERROR_KEY) != 0 ) {
+                getLogger().error("There were errors while inserting data documents {} result {}",
+                   new Object [] {documents, result});
+                session.transfer(flowFile, REL_FAILURE);
+            } else {
+                session.transfer(flowFile, REL_SUCCESS);
+                session.getProvenanceReporter().send(flowFile,
+                    new StringBuilder("rethinkdb://").append(databaseName).append("/").append(tableName).toString(),
+                    (endTimeMillis - startTimeMillis));
+            }
+        } catch (Exception exception) {
+            getLogger().error("Failed to insert into RethinkDB due to {}",
+                    new Object[]{exception.getLocalizedMessage()}, exception);
+            flowFile = session.putAttribute(flowFile, RETHINKDB_ERROR_MESSAGE, exception.getMessage() + "");
```

Comment:
I notice you use the `+ ""` method to cast objects to strings in several places.  Any special reason?  It does not appear too much in the NiFi codebase, as opposed to `String.valueOf()`.